platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java

```diff
@@ -48,52 +46,48 @@ public class TestFileUtils {
     private TestFileUtils() {}
 
     /**
-     * Create a new data file with specified initial contents.
+     * Create a new data file with specified initial contents or overwrite existing file.
      * @param f a file to create (parents will be created automatically)
      * @param body the complete contents of the new file (in UTF-8 encoding)
      */
     public static File writeFile(File f, String body) throws IOException {
         f.getParentFile().mkdirs();
-        OutputStream os = new FileOutputStream(f);
-        PrintWriter pw = new PrintWriter(new OutputStreamWriter(os, StandardCharsets.UTF_8));
-        pw.print(body);
-        pw.flush();
-        os.close();
+        Files.write(f.toPath(), body.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+        // TODO jdk11
+//        Files.writeString(f.toPath(), body, StandardOpenOption.CREATE_NEW);
         return f;
     }
 
     /**
      * Read the contents of a file as a single string.
-     * @param a data file
+     * @param file data file
      * @return its contents (in UTF-8 encoding)
      */
     public static String readFile(File file) throws IOException {
-        InputStream is = new FileInputStream(file);
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        byte[] buf = new byte[4096];
-        int read;
-        while ((read = is.read(buf)) != -1) {
-            baos.write(buf, 0, read);
-        }
-        is.close();
-        return baos.toString("UTF-8");
+        return new String(Files.readAllBytes(file.toPath()), StandardCharsets.UTF_8);
+        // TODO jdk11
+//        return Files.readString(file.toPath());
     }
 
     /**
      * Read the contents of a file as a byte array.
-     * @param a data file
+     * @param file data file
      * @return its raw binary contents
      */
     public static byte[] readFileBin(File file) throws IOException {
-        InputStream is = new FileInputStream(file);
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        byte[] buf = new byte[4096];
-        int read;
-        while ((read = is.read(buf)) != -1) {
-            baos.write(buf, 0, read);
+        return Files.readAllBytes(file.toPath());
+    }
+
+    /**
+     * Deletes the file or directory and its contents.
+     */
+    public static void deleteFile(File file) throws IOException {
+        if (file.isDirectory() && file.equals(file.getCanonicalFile())) {
```

Comment:
What is the reason for the second condition?

Reply:
I believe this is a way of checking that the recursion didn't walk through a symlink and is clearing half of the drive.

Please note that I copied this method mostly from the other util class (there were 15 variants of this method copied around). I tried to centralize it and made the actual delete operation use `Files#delete` to get a cause msg next time it fails in CI.

I thought about reimplementing this fully using `Path`, but keeping it as-is felt like the less risky approach for a hotfix. But now that I know that there is a shared test utility class, this opens up more cleanup and deduplicaiton in future.

Reply:
https://github.com/search?q=repo%3Aapache%2Fnetbeans%20%22file.isDirectory()%20%26%26%20file.equals(file.getCanonicalFile())%22&type=code more copies of this exact variant

Reply:
A thanks - kind of makes sense. Indeed I though, that `File.walk` might make more sense here.

Reply:
I will put this on my cleanup todo list

Reply:
or let me implement this properly right away, why not